prov/verbs/src/fi_verbs.h

```diff
@@ -435,6 +434,21 @@ struct fi_ibv_rdm_request;
 typedef void (*fi_ibv_rdm_cq_read_entry)(struct fi_ibv_rdm_request *cq_entry,
 					 int index, void *buf);
 
+static inline int
+fi_ibv_is_completion(uint64_t cq_flags, uint64_t op_flags)
+{
+	if ((op_flags & FI_COMPLETION) ||
+	    (op_flags & (FI_INJECT_COMPLETE	|
+			 FI_TRANSMIT_COMPLETE	|
+			 FI_DELIVERY_COMPLETE)))
+		return 1;
+	else if (op_flags & FI_INJECT)
+		return 0;
```

Comment:
Isn't this the exact same set of tests as before, with the same issue?

Reply:
Actually, no. The set of tests are extended by `FI_INJECT` test.

Reply:
This is a movement of fi_ibv_dgram_is_completion() without changes.  As @a-ilango points out, this looks incorrect.  FI_INJECT flag does not affect whether a completion is generated, only if the buffer is immediately reusable.  I think the correct test should be:
```
static inline int ofi_need_completion(cq_flags, op_flags)
{
    return !(cq_flags & FI_SELECTIVE_COMPLETION) ||
        (op_flags & (FI_COMPLETION | FI_INJECT_COMPLETE |
            FI_TRANSMIT_COMPLETE | FI_DELIVERY_COMPLETE);
}
```
The man pages aren't clear on how to handle FI_XXX_COMPLETE flags when selective completion is enabled.  I.e. must the FI_COMPLETION flag also be specified.  I went with your interpretation.

The above helper function should be moved into common code somewhere for all providers to use.  I didn't find anything equivalent in a quick search of the code.